packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -96,209 +97,1282 @@ export const handler = Sentry.AWSLambda.wrapHandler(
+                <a href="#vitals">Vitals</a>
+              </li>
+              <li>
+                <a href="#laboratory">Laboratory</a>
+              </li>
+            </div>
+            <div class='half'>
+              <li>
+                <a href="#encounters">Encounters</a>
+              </li>
+              <li>
+                <a href="#immunizations">Immunizations</a>
+              </li>
+              <li>
+                <a href="#family-member-history">Family Member History</a>
+              </li>
+              <li>
+                <a href="#related-persons">Related Persons</a>
+              </li>
+              <li>
+                <a href="#tasks">Tasks</a>
+              </li>
+              <li>
+                <a href="#coverage">Coverage</a>
+              </li>
+            </div>
+            </ul>
+        </div>
+      </div>
+    </div>
+  `;
 }
 
-async function getBundleFromS3(fileName: string) {
-  const getResponse = await s3Client
-    .getObject({
-      Bucket: bucketName,
-      Key: fileName,
+function createHeaderTableRow(label: string, value: string) {
+  return `
+    <tr>
+      <td>
+        <span class="header-label">${label}</span>
+        <span>
+          ${value}
+        </span>
+      </td>
+    </tr>
+  `;
+}
+
+function createMedicationSection(fhirBundle: Bundle) {
+  const medicationRequests: MedicationRequest[] | undefined = fhirBundle.entry
+    ?.filter(entry => {
+      return entry?.resource?.resourceType === "MedicationRequest";
     })
-    .promise();
-  const objectBody = getResponse.Body;
-  if (!objectBody) throw new Error(`No body found for ${fileName}`);
-  return JSON.parse(objectBody.toString());
+    .map(entry => {
+      return entry.resource as MedicationRequest;
+    });
+
+  if (!medicationRequests) {
+    return "";
+  }
+
+  const medicationsSortedByDate = medicationRequests.sort((a, b) => {
+    return dayjs(a.authoredOn).isBefore(dayjs(b.authoredOn)) ? 1 : -1;
+  });
+
+  const removeDuplicate = uniqWith(medicationsSortedByDate, (a, b) => {
+    const aDate = dayjs(a.authoredOn).format("YYYY-MM-DD");
+    const bDate = dayjs(b.authoredOn).format("YYYY-MM-DD");
+
+    return (
+      aDate === bDate && a.medicationCodeableConcept?.text === b.medicationCodeableConcept?.text
+    );
+  });
+
+  const blacklistInstructions = ["not defined"];
+
+  const medicalTableContents = `
+    <thead>
+      <tr>
+        <th style="width: 25%">Medication</th>
+        <th style="width: 25%">Instructions</th>
+        <div style="width: 50%">
+          <th>Dosage</th>
+          <th>Quantity</th>
+          <th>Refills</th>
+          <th>Status</th>
+          <th>Code</th>
+          <th>Date</th>
+        </div>
+      </tr>
+    </thead>
+    <tbody>
+      ${removeDuplicate
+        .map(medicationRequest => {
+          const code = getSpecificCode(medicationRequest.medicationCodeableConcept?.coding ?? [], [
+            "rxnorm",
+            "ndc",
```

Comment:
would suggest putting the medical codes in consts so we have a source of truth, and re-use accordingly